numpy/ma/core.py

```diff
@@ -3190,20 +3195,31 @@ def astype(self, newtype):
          [7 -- 9]]
 
         """
-        newtype = np.dtype(newtype)
-        newmasktype = make_mask_descr(newtype)
+        output = super(MaskedArray, self).astype(dtype,
+            order=order, casting=casting, subok=subok, copy=copy)
+
+        # if no copy was made, there's nothing to do
+        if output is self:
+            return self
+
+        # asked to return a base array, so nothing to decorate
+        if not subok:
+            return output
```

Comment:
Yeah, I was unsure of that too - not clear to me how best to apply liskov substitution here. Only did it this way because it was slightly easier